crates/hyperqueue/src/dashboard/ui/screen/mod.rs

```diff
@@ -1,13 +1,43 @@
+use tako::WorkerId;
 use termion::event::Key;
+use tui::layout::Rect;
 
 use crate::dashboard::data::DashboardData;
 use crate::dashboard::ui::screen::controller::ScreenController;
 use crate::dashboard::ui::terminal::DashboardFrame;
 
 pub mod controller;
 
+pub trait Fragment {
+    fn draw(&mut self, in_area: Rect, frame: &mut DashboardFrame);
+    fn update(&mut self, data: &DashboardData, controller: &mut ScreenController);
+    fn handle_message(&mut self, message: ToFragmentMessage);
+    fn handle_key(
+        &mut self,
+        key: Key,
+        controller: &mut ScreenController,
+    ) -> Option<FromFragmentMessage>;
```

Comment:
Why is this so general? I would expect that each fragment is "private" for a screen and it can generate different events. If we standardize the messages for all fragments, then the screens would have to support all possible fragment messages.

Reply:
> Why is this so general? I would expect that each fragment is "private" for a screen and it can generate different events. If we standardize the messages for all fragments, then the screens would have to support all possible fragment messages.

I wanted to be able to re-use fragments across screens. I guess one possible solution to having separate messages for the screens would be using a generic in the Fragment trait for the type of message and having different messages for each of the screens, what do you think?